File: src/main/java/pl/engineerproject/pw/fifapp/service/LeagueService.java
package pl.engineerproject.pw.fifapp.service;

import pl.engineerproject.pw.fifapp.dto.LeagueDto;

import java.util.List;

public interface LeagueService {
    LeagueDto getLeagueById(Integer leagueId);
    void saveLeague(LeagueDto leagueDto);
    List<LeagueDto> getAllLeagues();
    void deleteLeague(Integer leagueId);
    void updateLeague(LeagueDto leagueDto);

}

File: src/main/resources/static/webapp/League/LeagueService.js
'use strict'

angular.module('league.services', []).service('LeagueService', ["$http", function($http) {

    this.getAllLeagues = function getAllLeagues() {
        return $http({
            method: 'GET',
            url: 'leagues/getAllLeagues'
        });
    }

    this.addLeague = function addLeague(name, startDate, endDate, description, location) {
        return $http({
            method: 'POST',
            url: 'leagues/saveLeague',
            data: {
                name: name,
                description: description,
                startDate: startDate,
                endDate: endDate,
                location: location
            }
        });
    }

    this.updateLeague = function updateLeague(leagueId, name, startDate, endDate, description, location) {
        return $http({
            method: 'PUT',
            url: 'leagues/update',
            data: {
                leagueId: leagueId,
                name: name,
                description: description,
                startDate: startDate,
                endDate: endDate,
                location: location
            }
        })
    }

    this.deleteLeague = function deleteLeague(leagueId) {
        return $http({
            method: 'DELETE',
            url: 'leagues/delete/' + leagueId
        })
    }



}]);